plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListenerModule.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage;
+
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.openlineage.client.OpenLineageClient;
+import io.openlineage.client.transports.ConsoleTransport;
+import io.openlineage.client.transports.HttpConfig;
+import io.openlineage.client.transports.HttpTransport;
+import io.openlineage.client.transports.TokenProvider;
+import io.trino.plugin.openlineage.config.OpenLineageClientHttpTransportConfig;
+import io.trino.plugin.openlineage.config.OpenLineageListenerConfig;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static java.lang.String.format;
+
+public class OpenLineageListenerModule
+        extends AbstractConfigurationAwareModule
+{
+    private static class ApiKeyTokenProvider
+            implements TokenProvider
+    {
+        private final String token;
+
+        public ApiKeyTokenProvider(String token)
+        {
+            this.token = token;
+        }
+
+        @Override
+        public String getToken()
+        {
+            return format("Bearer %s", this.token);
+        }
+    }
+
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(OpenLineageListener.class).in(Scopes.SINGLETON);
+
+        configBinder(binder).bindConfig(OpenLineageListenerConfig.class);
+
+        newOptionalBinder(binder, OpenLineageClientHttpTransportConfig.class);
+
+        install(conditionalModule(
+                OpenLineageListenerConfig.class,
+                config -> config.getTransport()
+                        .equals(OpenLineageTransport.HTTP),
+                conditionalBinder -> configBinder(conditionalBinder).bindConfig(OpenLineageClientHttpTransportConfig.class)));
```

Comment:
We could create a new module which could capture the changes related to `HTTP` transport.